plugins/lsp/lsp-status.nix

```
@@ -0,0 +1,82 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "lsp-status";
+  originalName = "lsp-status.nvim";
+  defaultPackage = pkgs.vimPlugins.lsp-status-nvim;
+  maintainers = [ helpers.maintainers.b3nb5n ];
+  settingsOptions =
+    let
+      indicator_opt = helpers.defaultNullOpts.mkStr "" ''
+        string to show as diagnostics.
+        If you don't have Nerd/Awesome Fonts you can replace defaults with ASCII chars.
+      '';
+    in
+    {
+      kind_labels = helpers.defaultNullOpts.mkAttrsOf types.str "" ''
+        An optional map from LSP symbol kinds to label symbols. Used to decorate the current function name.
+      '';
+
+      select_symbol = helpers.defaultNullOpts.mkStr "" ''
+            An optional handler of the form function(cursor_pos, document_symbol) that should return true if document_symbol (a DocumentSymbol) should be accepted as the symbol currently containing the cursor.
+        	'';
+
+      current_function = helpers.defaultNullOpts.mkBool true ''
+        True if the current function should be updated and displayed in the default statusline component.
+      '';
+
+      show_filename = helpers.defaultNullOpts.mkBool true ''
+        True if the current function should be updated and displayed in the default statusline component.
+      '';
+
+      indicator_ok = indicator_opt;
+      indicator_errors = indicator_opt;
+      indicator_warnings = indicator_opt;
+      indicator_info = indicator_opt;
+      indicator_hint = indicator_opt;
+
+      indicator_separator = helpers.defaultNullOpts.mkStr " " ''
+        A string which goes between each diagnostic group symbol and its count.
+      '';
+
+      component_separator = helpers.defaultNullOpts.mkStr " " ''
+        A string which goes between each "chunk" of the statusline component (i.e. different diagnostic groups, messages).
+      '';
+
+      diagnostics = helpers.defaultNullOpts.mkBool true ''
+        If false, the default statusline component does not display LSP diagnostics.
+      '';
+    };
+
+  callSetup = false;
+  extraConfig = cfg: {
+    assertions = [
+      {
+        assertion = cfg.plugins.lsp.enable;
```

```suggestion
        assertion = config.plugins.lsp.enable;
```

`cfg` is _this plugin's_ part of the config